packages/builtin/string.pony

```diff
@@ -101,14 +100,36 @@ actor Main
       _set(0, 0)
     else
       _size = len
+      _alloc = alloc.max(_size + 1)
+      _ptr = str
+    end
 
-      if len == 0 then
-        while str._apply(_size) != 0 do
-          _size = _size + 1
+  new from_cstring(str: Pointer[U8]) =>
+    """
+    Return a string from a pointer to a null-terminated cstring
+    without making a copy. The data is not copied. This must be done
+    only with C-FFI functions that return pony_alloc'd character
+    arrays. The pointer is scanned for the first null byte, which will
+    be interpreted as the null terminator. If a null pointer is given
+    then an empty string is returned.
```

Comment:
I think the docstring should warn about the dangers of passing in a pointer that has no null byte within its bounds.  Specifically, it seems that doing so would violate memory safety.  Obviously this can only be done via FFI, but it should still be documented.  The previous implementation of `from_cstring` also gave a warning about memory safety if the `len` argument was not given, and there was no null byte in the scan.